### DISCUSSION TITLE
Is importing aggregatorV3interface in mockV3aggregator necessary?

### DISCUSSION THREAD

**Discussion by Fahad3242** [Cyfrin/foundry-full-course-cu#3142] [Q&A]
Can anyone explain me why import was added in MockV3Aggregator 
![image](https://github.com/user-attachments/assets/385aa0bf-85be-4ace-91ab-b074da532dc8)

When I removed it worked fine
![image](https://github.com/user-attachments/assets/75f33091-ac17-4d71-a76c-51596c388987)

THANKS IN ADVANCE

**Comment by EngrPips** [comment]
Hello @Fahad3242, We always make a contract an interface type so that we ensure that we implement all the functions and features we have planned out without forgetting anyone because the contract will not compile if you did not completely implement everything outlined in the interface. Removing the interface might lead to a scenario where you forget to implement some stuff you planned to.

Please ask questions if anything is confusing.

**Reply by Fahad3242** [reply]
Thank you for the clarification. Why is it giving an error when I make the contract agginterface type

**Reply by EngrPips** [reply]
Because you did not properly import the Interface, It explicitly informed you in the error.

**Reply by Fahad3242** [reply]
I just copied the mockV3agg it gave me that import error again which I faced last time when importing AggregatorV3interface. The file is exactly where the destination is written in the import line.

**Reply by EngrPips** [reply]
I will advise you to check correctly.
